final_task/rss_reader/action_functions.py

```diff
@@ -0,0 +1,195 @@
+    parser.add_argument("--limit", type=int, help="Limit news topics if this parameter provided")
+
+    return parser.parse_args()
+
+
+def get_news(command_line_args, logger):
+    """ Get news function.
+
+        Uses feedparser library to receive news,
+        and BeautifulSoup library to converting news in readable format.  """
+    logger.info("Getting news.")
+    news_feed = feedparser.parse(command_line_args.source)
+
+    # convert title string to unicode
+    feed = {"title": html.unescape(news_feed.feed.get("title", "")),
+            "language": news_feed.feed.get("language", "")}
+    news_collection = {"feed": feed,
+                       "entries": []}
+    source = command_line_args.source
+
+    for entry in news_feed.entries:
+        news_entry = NewsEntry()
+        news_entry.feed_title = feed["title"]
+        news_entry.feed_language = feed["language"]
+
+        news_entry.source = source
+        news_entry.title = html.unescape(entry.get("title", ""))
+        news_entry.date = entry.get("published", "")
+        news_entry.link = entry.get("link", "")
+
+        # get rid of html tags
+        soup = BeautifulSoup(entry.get("summary", ""), "html.parser")
+        news_entry.summary = html.unescape(soup.text)
+        # get images links
+        images = soup.findAll("img")
+        #mayby add images_alt text in future
+        for img in images:
+            news_entry.image_links.append(img["src"])
+
+        news_collection["entries"].append(news_entry)
+
+    return news_collection
+
+
+def print_news_stdout(news_collection):
+    """ Function for print news to stdout in text format. """
+    print("################################################################################\n",
+          "Feed: " + news_collection["feed"]["title"],
+          "Language: " + news_collection["feed"]["language"] + '\n',
+          sep='\n')
+
+    for entry in news_collection["entries"]:
+        entry.print_entry()
+
+
+def print_news_json(news_collection):
+    """ Function for print news to stdout in json format. """
+    news_collection_for_json = {"feed": news_collection["feed"],
+                                "entries": []}
+
+    for entry in news_collection["entries"]:
+        entry_for_json = asdict(entry)
+        news_collection_for_json["entries"].append(entry_for_json)
+
+    print(json.dumps(news_collection_for_json, indent=4))
+
+
+def print_news(news_collection, com_line_args, logger):
+    """ Function for print news to stdout,
+        that take account of limit and json arguments. """
+
+    # get valid limit argument
+    # if not initialize limit argument
+    if not check_limit_arg(com_line_args, logger):
+        limit = len(news_collection["entries"])
+    else:
+        limit = com_line_args.limit
+
+    if len(news_collection["entries"]) < limit:
+        logger.warning("The number of news is less than the value of the argument limit.")
```

Comment:
To my mind, it's not the warning that user may want to see. For example, we have only 3 news and the limit is 5. It's completly ok. You could notify user, available news > chosen limit. But this is optional

Reply:
This warning only for me, users can see only errors or higher level logs.